src/vs/workbench/browser/parts/quickopen/quickOpenController.ts

```diff
@@ -549,6 +568,33 @@ export class QuickOpenController extends WorkbenchComponent implements IQuickOpe
 		return promiseCompletedOnHide;
 	}
 
+	private handleOnCancel(isPicker: boolean): void {
+		// restore the editor part state after cancelling
+		this.historyService.block(true);
+
+		// restore the previous preview editor
+		if (this.previousPreviewEditorInput) {
```

Comment:
@wprater unfortunately I think this is more complicated for the restore part. I can see the following states:
* 1.) no active editor
* 2.) no preview editor
* 3.) active editor is preview editor
* 4.) active editor is not preview editor

1.) easy, just close the preview editor that was opened
2.) also easy, just close the preview editor
3.) restore the previous active editor by opening it
4.) restore the previous editor in the preview slot but open it in the background because it was not the active one. also open the previous active editor

Also you have to be careful that you do this from the `onCancel()` callback. Because as far as I remember this method is not called when you actually pick an entry from quick open. I would expect some additional logic when an element is picked.